doc/source/dev/distributing.rst

```diff
@@ -0,0 +1,180 @@
+Distributing
+============
+
+Distributing Python packages is nontrivial - especially for a package with
+complex build requirements like Scipy - and subject to change.  For an up-to-date
+overview of recommended tools and techniques, see the `Python Packaging User
+Guide`_.  This document discusses some of the main issues and considerations for
+Scipy.
+
+Dependencies
+------------
+Dependencies are things that a user has to install in order to use (or
+build/test) a package.  They usually cause trouble, especially if they're not
+optional.  Scipy tries to keep its dependencies to a minimum; currently they
+are:
+
+*Unconditional run-time dependencies:*
+- Numpy_
+
+*Conditional run-time dependencies:*
+- nose_ (to run the test suite)
+- asv_ (to run the benchmarks)
+- matplotlib_ (for some functions that can produce plots)
+- Pillow_ (for image loading/saving)
+- scikits.umfpack_ (optionally used in ``sparse.linalg``)
+- mpmath_ (for more extended tests in ``special``)
+
+*Unconditional build-time dependencies:*
+- Numpy_
+- A BLAS and LAPACK implementation (reference BLAS/LAPACK, ATLAS, OpenBLAS,
+  MKL, Accelerate are all known to work)
+- (for development versions) Cython_
+
+*Conditional build-time dependencies:*
+- setuptools_
+- wheel_ (``python setup.py bdist_wheel``)
+- Sphinx_ (docs)
+- matplotlib_ (docs)
+- LaTeX (pdf docs)
+
+Furthermore of course one needs C, C++ and Fortran compilers to build Scipy,
+but those we don't consider to be dependencies and are therefore not discussed
+here.  For details, see http://scipy.org/scipylib/building/index.html.
+
+When a package provides useful functionality and it's proposed as a new
+dependency, consider also if it makes sense to vendor (i.e. ship a copy of it with
+scipy) the package instead.  For example, six_ and decorator_ are vendored in
+``scipy._lib``.
+
+The only dependency that is reported to pip_  is Numpy_, see
+``install_requires`` in Scipy's main ``setup.py``.  The other dependencies
+aren't needed for Scipy to function correctly, and the one unconditional build
+dependency that pip_ knows how to install (Cython_) we prefer to treat like a
+compiler rather than a Python package that pip_ is allowed to upgrade.
+
+Issues with dependency handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are some serious issues with how Python packaging tools handle
+dependencies reported by projects.  Because Scipy gets regular bug reports
+about this, we go in a bit of detail here.
+
+Scipy only reports its dependency on Numpy via ``install_requires`` if Numpy
+isn't installed at all on a system.  This will only change when there are
+either 32-bit and 64-bit Windows wheels for Numpy on PyPi or when
+``pip upgrade`` (with sane behavior, see `this PR
```

Comment:
or when ``pip upgrade`` behavior changes (see ...) [need a verb for the second clause]